open3d/io/file_format/FileGLTF.cpp

```diff
@@ -814,28 +811,21 @@ bool SaveMeshGLTF(const std::string &fileName, const geometry::TriangleMesh &_me
       extendBufferConvert<Eigen::Vector2f>(mesh.triangle_uvs_, gltfBuffer, vertexTexcoordBufferView.byteOffset, vertexTexcoordBufferView.byteLength);
       gltfModel.bufferViews.emplace_back(std::move(vertexTexcoordBufferView));
       // setup material
-      auto setup_texture = [&](const geometry::Image &image, const std::string &temporary_file_name, auto &texture_info) {
-        texture_info.index = gltfModel.textures.size();
-        texture_info.texCoord = 0;
-        tinygltf::Texture texture;
-        texture.source = gltfModel.images.size();
-        gltfModel.textures.emplace_back(std::move(texture));
-        add_image(image, temporary_file_name);
+      auto setup_texture_if_needed = [&](const std::optional<unsigned int> &texture_index, auto &texture_info) {
+        if (texture_index.has_value()) {
+          texture_info.index = *texture_index;
+          texture_info.texCoord = 0;
+        }
       };
-      if (material.gltfExtras.texture_idx.has_value()) {
-        setup_texture(mesh.textures_[*material.gltfExtras.texture_idx], "texture.jpg", gltfMaterial.pbrMetallicRoughness.baseColorTexture);
-      } else if (material.albedo) {
-        setup_texture(*material.albedo, "texture.jpg", gltfMaterial.pbrMetallicRoughness.baseColorTexture);
-      }
-      if (material.normalMap) {
-        setup_texture(*material.normalMap, "normal.jpg", gltfMaterial.normalTexture);
-      }
-      if (material.ambientOcclusion) {
-        setup_texture(*material.ambientOcclusion, "occlusion.jpg", gltfMaterial.occlusionTexture);
-      }
-      if (material.roughness) {
-        setup_texture(*material.roughness, "roughness.jpg", gltfMaterial.pbrMetallicRoughness.metallicRoughnessTexture);
-      }
+
+      setup_texture_if_needed(material.gltfExtras.texture_idx.has_value() ? material.gltfExtras.texture_idx : material.albedo,
+                              gltfMaterial.pbrMetallicRoughness.baseColorTexture);
+      setup_texture_if_needed(material.normalMap, gltfMaterial.normalTexture);
+      setup_texture_if_needed(material.ambientOcclusion, gltfMaterial.occlusionTexture);
+      setup_texture_if_needed(material.roughness, gltfMaterial.pbrMetallicRoughness.metallicRoughnessTexture);
+      setup_texture_if_needed(material.gltfExtras.emissiveTexture, gltfMaterial.emissiveTexture);
+      //! @note The following textures are ignored: metallic, reflectance, clearCoat, clearCoatRoughness and anisotropy because there is nothing
```

Comment:
Will this list of different things (metallic, reflectance, etc.) that is repeated throughout these edits, change at any point? It seems like a lot to keep track of and make sure the list of these things is always accounted for in each place.

Reply:
Is there a better way to keep track of everything all in one place, or are we bound by how things already work?

Reply:
Most likely it won't change. This should be everything in the gltf format which is pretty stable. If we start having full support for other file formats it might change on the `Material` data structure, but not in the GLTF reader/writer. I don't think there's a better way to deal with this. I already added the `setup_texture_if_needed()` lambda to get rid of copy and pasted repeated logic.

Reply:
Yea I saw that you did that. Makes sense!